chain/eth_transaction_hash_lookup.go

```diff
@@ -0,0 +1,162 @@
+package chain
+
+import (
+	"database/sql"
+	"math"
+
+	"github.com/ipfs/go-cid"
+	_ "github.com/mattn/go-sqlite3"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+var pragmas = []string{
+	"PRAGMA synchronous = normal",
+	"PRAGMA temp_store = memory",
+	"PRAGMA mmap_size = 30000000000",
+	"PRAGMA page_size = 32768",
+	"PRAGMA auto_vacuum = NONE",
+	"PRAGMA automatic_index = OFF",
+	"PRAGMA journal_mode = WAL",
+	"PRAGMA read_uncommitted = ON",
+}
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tx_hash_lookup (
+		hash TEXT PRIMARY KEY,
+		cid TEXT NOT NULL,
+		epoch INT NOT NULL
+	)`,
```

Comment:
Don't we want some indexes here?
* On hash for lockups by hash
* On CID by lockups by cid
* Sorted index on epoch for efficient range delete

Something like
```sql
create table if not exists tx_hash_lookup
(
    hash  text    not null
        primary key
        unique,
    cid   text    not null
        unique,
    epoch integer not null
);

create unique index if not exists tx_hash_lookup_cid_uindex
    on tx_hash_lookup (cid);

create index if not exists tx_hash_lookup_epoch_index
    on tx_hash_lookup (epoch);

create unique index if not exists tx_hash_lookup_hash_uindex
    on tx_hash_lookup (hash);
```

Reply:
Unique colums (including primary key) get indexed automatically. I've added an index for the `insertion_time`, which has replaced the epoch for garbage collection purposes.